packages/flutter_tools/lib/src/runner/local_engine.dart

```diff
@@ -113,12 +76,79 @@ class LocalEngineLocator {
     if (engineSourcePath != null) {
       _logger.printTrace('Local engine source at $engineSourcePath');
       return _findEngineBuildPath(localEngine, engineSourcePath);
-    } else if (localEngine != null) {
-      _logger.printTrace('Could not find engine source for $localEngine, skipping');
+    }
+    if (localEngine != null) {
+      throwToolExit(
```

Comment:
Now throw instead of silently fail if `--local-engine` is set but the `engineSourcePath` can't be found.